File: app/src/main/java/com/exercise/davismiyashiro/popularmovies/movies/CustomBindingAdapters.java
package com.exercise.davismiyashiro.popularmovies.movies;

import androidx.databinding.BindingAdapter;
import androidx.recyclerview.widget.RecyclerView;
import android.widget.ImageView;

import com.exercise.davismiyashiro.popularmovies.moviedetails.MovieDetailsObservable;
import com.squareup.picasso.Picasso;

import java.util.List;

/**
 * Created by Davis Miyashiro.
 */

public class CustomBindingAdapters {

    public static final String IMG_BASE_URL = "https://image.tmdb.org/t/p/w500";

    @BindingAdapter({"imgUrl"})
    public static void loadImage(ImageView view, String url) {
        if (url!= null && !url.equals("")) {
            Picasso.with(view.getContext())
                    .load(IMG_BASE_URL + url)
                    .placeholder(android.R.drawable.progress_indeterminate_horizontal)
                    .into(view);
        }
    }

    @BindingAdapter({"app:data"})
    public static void loadMovieDetails(RecyclerView recyclerView, List<MovieDetailsObservable> movies) {
        if (recyclerView.getAdapter() instanceof MovieListAdapter) {
            ((MovieListAdapter)recyclerView.getAdapter()).replaceData(movies);
        }
    }
}

File: app/src/main/java/com/exercise/davismiyashiro/popularmovies/movies/CustomBindingAdapters.kt
/*
 * MIT License
 *
 * Copyright (c) 2019 Davis Miyashiro
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.exercise.davismiyashiro.popularmovies.movies

import android.widget.ImageView
import androidx.databinding.BindingAdapter
import com.squareup.picasso.Picasso

/**
 * Created by Davis Miyashiro.
 */

object CustomBindingAdapters {

    private const val IMG_BASE_URL = "https://image.tmdb.org/t/p/w500"

    @BindingAdapter("imgUrl")
    @JvmStatic
    fun loadImage(view: ImageView, url: String?) {
        url?.let {
            Picasso.get()
                .load(IMG_BASE_URL + it)
                .placeholder(android.R.drawable.progress_indeterminate_horizontal)
                .into(view)
        }
    }
}
